baybe/utils/plotting.py

```diff
@@ -0,0 +1,78 @@
+"""Plotting utilities."""
+
+import json
+import os
+import sys
+import warnings
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
```

Comment:
What is our design pattern now on optional dependencies? 
seaborn and pyplot could be lazily imported as they are strictly speaking optional dependencies of `[examples]`

But im not sure if thats required anymore since `import from utils` top level was deactivated

Reply:
I do not know if we actually already have an agreed-upon global design pattern for this kind of stuff. However, lazily importing here makes sense as it for example prevents the import from happening when being run with `SMOKE_TEST` being set, so I will change that.

Reply:
Lazy loading is now included in the sense that most stuff is only loaded when it is needed (grouped the imports together to improve readability). If you are happy with this, please resolve, otherwise I can also drop the commit again.

Reply:
Note: Made one error here with the import of `warning`, will have to fix that

Reply:
no need to shift all other packages too, i was jsut thinking about the optioal ones

if we start to shift all other packages to we will have to rebase the entire package :D 🗡️

Reply:
Well, but then it is completely arbitrary which packages to shift and which not to shift? And what qualifies as "optional" anyway, here `warnings` for example is optional in that it is not necessarily needed. So in that case, I'd prefer to not choose some arbitrary list of packages, declare these to be "optional" and import them lazily, specifically if there is not good reason like we have e.g. with `torch`.
And don't mention anything about doing a full rebase, or thou might awaken the Godfather of refactorings from his deep slumber ;)

Reply:
TL;DR: Is just not using lazy import here gucci for you? Can still change this at a later point. Then I'd change.

Reply:
optional is what is not available when not intalled with the optional dependency, matplotlib and seaborn are not installed, only if `baybe[examples]`. warnings, sys, os and such comes with every python installation

If you now shift non-optional dependencies to lazy import, why do we not do that in all of the code?

Reply:
Ok, that makes sense. Will change the lazy imports for matplotlib and seaborn and restore the other parts and won't shift the others.

Reply:
Done. Please resolve if gucci.